src/pages/webinars/code-insights-turning-your-metrics-into-action/watch-now.tsx

```diff
@@ -0,0 +1,62 @@
+        img: '/staff/joel-kwartler.png',
+        bio: 'Joel is the Product Manager for Code Insights at Sourcegraph, and wants to bring the best practices of big data to the problems of big code. Joel has bounced around the startup + design world at places like Figma and IDEO, and got a bachelor’s in C.S. at Harvard. After hours, you might find him doing stand up, pretending he knows piano, or waking up way too early to take photographs. He’s made the Forbes 30 Under 30 List of Best Joel Kwartlers for 5 of the past 8 years.',
+    },
+    {
+        name: 'Shawn King',
+        title: 'Sourcegraph Customer Engineer',
+        img: '/staff/shawn-king.png',
+        bio: 'Shawn is a customer engineer at Sourcegraph. He has helped people use technology to get insights into their work his entire career. Coming from Microsoft and a startup called Intelligent InSites, Shawn has been able to support and provide metrics to many customers across different industries. Between the times when he’s helping customers, you can catch Shawn reading science fiction and enjoying time with his family.',
+    },
+]
+
+const Recording: FunctionComponent = () => (
+    <Layout
+        meta={{
+            title: 'Sourcegraph Code Insights: Turning Metrics into Action',
+            description:
+                'Join Sourcegraph Product Manager Joel Kwartler and Customer Engineer Shawn King to learn how to use Code Insights to turn your most pressing analytics questions into KPIs you can track with mere seconds of setup.',
+        }}
+    >
+        <GatedResourceLayout
+            title={
+                <>
+                    Sourcegraph Code Insights:
+                    <br /> Turning Metrics into Action
+                </>
+            }
+            subtitle="How to track what really matters to you and your team"
+            speakers={speakers}
+            videoSrc="https://www.youtube.com/embed/dXKvetMozB0"
+            learnMoreCTA={
+                <ContentSection className="d-flex flex-column align-items-center py-lg-8 py-7">
+                    <h1 className="font-weight-bold text-center">Want to learn more about Code Insights?</h1>
+                    <Link href="/contact/request-code-insights-demo" passHref={true}>
+                        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+                        <a
+                            className="btn btn-primary mt-4 col-12 col-md-3 col-xl-2"
+                            title="Request a Demo"
+                            data-button-style={buttonStyle.primary}
+                            data-button-location={buttonLocation.bodyDemo}
+                            data-button-type="cta"
+                        >
+                            Request a demo
+                        </a>
+                    </Link>
+                </ContentSection>
+            }
+        />
+    </Layout>
+)
+
+export default Recording
```

Comment:
Instead of creating a duplicate page, we can avoid some duplicate content here and I think we can optimize the `GatedResourceLayout` component to use a `?watch-now` query parameter. Then you can use that to conditionally show the `videoSrc` and swap that content. You can also use the param condition to not run the `useHubSpot` hook which will remove the need for an optional `formId`. This page with or without the `watch-now` parameter can be measured against each other for analytics.

Reply:
I see what you mean. I like the query param path better, less redundant, slightly different slug but better practice. Although, the hook will still run in the background when the form portion is hidden because we can't conditionally call hooks as part of the [React hook rules](https://reactjs.org/docs/hooks-rules.html). I tried disabling the linter to do this, but then it won't compile. Do you have another idea to exit the hook when it's not needed? Or think it's fine to leave running?

Reply:
Ah, I forgot about that hook rule for conditionals. In that case, I don't think we should leave the hook running since it would load for folks directly navigating to the resource.

I was going to suggest using `useState` and passing that to `onFormSubmitted` in the `GatedResourceLayout` so you could do something like `hasWatchNowQuery || formSubmitted` instead of `router.push`, but then that would prevent us from tracking those analytics between the pages for users submitting forms. But we'd still run into the hook being fired for users navigating directly to it.

My apologies, I didn't think this through thoroughly. I think we may have to revert to your previous solution but with a few optimizations since the URL is cleaner with `page-name/watch-now`. I'd suggest sharing the duplicated data between the two pages so you can add them into each `GatedResourceLayout` for each page conveniently. You could export the data from one page, and import in the `watch-now` page.